src/query/atomWithInfiniteQuery.ts

```diff
@@ -166,7 +200,14 @@ export function atomWithInfiniteQuery<
         }
       }
 
-      const defaultedOptions = queryClient.defaultQueryObserverOptions(options)
+      const defaultedOptions = queryClient.defaultQueryOptions(
+        options
+      ) as unknown as InfiniteQueryObserverOptions<
```

Comment:
How would types break without this assertion?

Reply:
Thanks. Why was it working previously? Do you think it's the issue of RQ?

Reply:
Because i passed QueryKey to InfiniteQueryObserverOptions, so that returns a type that contains QueryKey, but InfiniteQueryObserver's constructor that can not accept a options that contains QueryKey

Reply:
> can not accept a options that contains QueryKey

That sounds unlikely unless it a tricky type. Can you then use `Omit<>` to remove `QueryKey`? I really hope to remove `as unknown`.

Reply:
I tried, but its impossible. maybe u can wait for this issue to be merged and published then remove this trick
https://github.com/tannerlinsley/react-query/pull/3571